text/1002-required-by-default.rst

```diff
@@ -0,0 +1,178 @@
+..
+    Status: Draft
+    Type: Guideline
+    Created: 2020-06-17
+    RFC PR: `edgedb/rfcs#0001 <https://github.com/edgedb/rfcs/pull/1>`_
+
+===============================================================
+RFC 1002: REQUIRED qualifier by default in properties and links
+===============================================================
+
+This RFC describes the reasons why DDL and SDL declarations for properties
+and links assume non-optionality by default.
+
+
+Motivation
+==========
+
+The data definition language, and more importantly, the schema definition
+language, which users use to communicate with EdgeDB, need to avoid
+unnecessary verbosity.
+
+This requires some conventions and assumptions about what the user expects
+most of the time. Those assumptions shouldn't bar the user from changing
+their mind when the need arises.
+
+A default value has far reaching consequences which we cover in this
+document.
+
+
+Implementation
+==============
+
+Since the discussion always applies to properties and links, we're referring
+to them simply as *fields* below.
+
+Assume REQUIRED qualifiers when no explicit qualifier is provided
+-----------------------------------------------------------------
+
+The main advantage of assuming that all fields are required unless otherwise
+specified is its resilience to migration into optionality later. Client code
+that was forced to provide values before will continue working and there are
+no incompatible empty sets in the database prior to the migration.
+
+An extreme form of optionality everywhere is equivalent to the database
+not having a true schema. Unless it's necessary to make a field optional,
+forcing fields to be required ensures that whenever data *can* be specified,
+it won't be forgotten at INSERT or UPDATE time.
+
+Required fields simplify type information. There's two particular problems
```

Comment:
This argument would hold better if we disallowed `OPTIONAL` altogether and went full-on algebraic types. I don't think that's realistic, so in real schemas a substantial number of pointers would still be defined as `OPTIONAL` and so these problems would still apply.